src/process/receipts.rs

```diff
@@ -0,0 +1,170 @@
+                    receipt_action_input_data.extend(input_data_ids.iter().map(|data_id| {
+                        models::ReceiptActionInputData::from_data_id(
+                            receipt.receipt_id.to_string(),
+                            data_id.to_string(),
+                        )
+                    }));
+                    receipt_action_output_data.extend(output_data_receivers.iter().map(
+                        |receiver| {
+                            models::ReceiptActionOutputData::from_data_receiver(
+                                receipt.receipt_id.to_string(),
+                                receiver,
+                            )
+                        },
+                    ));
+                    receipt_action_actions.push(models::ReceiptActionAction::from_action_view(
+                        receipt.receipt_id.to_string(),
+                        i32::from_usize(index).unwrap(),
+                        action,
+                    ));
+                }
+            }
+        }
+    }
+
+    match diesel::insert_into(schema::receipt_actions::table)
+        .values(receipt_actions)
+        .execute_async(&pool)
+        .await
+    {
+        Ok(_) => (),
+        Err(async_error) => {
+            error!(target: "indexer_for_explorer", "Error occurred while ReceiptActions were saving... \n {:#?}", async_error);
+        }
+    };
+
+    match diesel::insert_into(schema::receipt_action_actions::table)
+        .values(receipt_action_actions)
+        .execute_async(&pool)
+        .await
+    {
+        Ok(_) => (),
+        Err(async_error) => {
+            error!(target: "indexer_for_explorer", "Error occurred while ReceiptActionActions were saving... \n {:#?}", async_error);
+        }
+    };
+
+    match diesel::insert_into(schema::receipt_action_output_data::table)
+        .values(receipt_action_output_data)
+        .execute_async(&pool)
+        .await
+    {
+        Ok(_) => (),
+        Err(async_error) => {
+            error!(target: "indexer_for_explorer", "Error occurred while ReceiptActionOutputData were saving... \n {:#?}", async_error);
+        }
+    };
+
+    match diesel::insert_into(schema::receipt_action_input_data::table)
+        .values(receipt_action_input_data)
+        .execute_async(&pool)
+        .await
+    {
+        Ok(_) => (),
+        Err(async_error) => {
+            error!(target: "indexer_for_explorer", "Error occurred while ReceiptActionInputData were saving... \n {:#?}", async_error);
+        }
+    };
+}
+
+async fn process_receipt_data(
+    pool: &Pool<ConnectionManager<PgConnection>>,
+    receipts: &[&near_indexer::near_primitives::views::ReceiptView],
+) {
+    let receipt_data_models: Vec<models::ReceiptData> = receipts
+        .iter()
+        .filter_map(|receipt| match models::ReceiptData::try_from(*receipt) {
+            Ok(model) => Some(model),
+            Err(_) => None,
+        })
```

Comment:
```suggestion
        .filter_map(|receipt| models::ReceiptData::try_from(*receipt).ok())
```

Reply:
I knew there should be an easier way, thanks